python_modules/dagster/dagster/core/definitions/partition.py

```diff
@@ -759,6 +768,23 @@ def get_run_config(self, partition_key: str) -> Dict[str, Any]:
             )
         return self.run_config_for_partition_fn(matching[0])
 
+    def get_tags(self, partition_key: str) -> Dict[str, Any]:
+        matching = [
```

Comment:
I see that we're already doing this in `get_run_config`, but it feels dicey to me that this is an O(# partitions) operation. Partition sets with 10k partitions - every hour for a couple years - is a common use case.

Reply:
Also, it looks like one code path use this method, while another code path uses the `tags_for_partition_fn` property. Without digging into the implementations, it would be hard for me to guess when one is appropriate vs. the other.

Reply:
This is the downside of having a `Partition` object vs a string key as the base API.  The partition set only knows how to produce an ordered list of `Partition` objects.  It does not know how to produce a `Partition` object for a string key.  The decorated function returns a set of tags for some generic partition, so we need to know how to resolve it.

It's actually really annoying to deal with generic `Partition` objects.  We have to deal with them for time window and dynamic based partition defs.  Also, since they might not be serializable, we have to resolve all the config/tags off of these definitions.

I'd actually be thrilled if we simplified the API and resolved everything to a string type, but I think that's a breaking change.

Reply:
Actually, I'll just get rid of these functions.  Nothing is using them except the internal job definition.

The `O(# of partitions)` issue will still exist but we won't do it twice in a row.  To get this down to constant time will require much bigger API changes.